addons/HotelPvP/scripts/Game/AFM_HotelPvP/Editor/Components/Editor/AFM_CameraLimitedEditorComponent.c

```diff
@@ -0,0 +1,84 @@
+modded class SCR_CameraLimitedEditorComponent : SCR_CameraEditorComponent
+{
+	override protected ResourceName GetCameraPrefab()
+	{
+		SCR_EditorManagerEntity manager = SCR_EditorManagerEntity.GetInstance();
+		if (!manager) 
+			return ResourceName.Empty;
+		
+		PlayerController pc = GetGame().GetPlayerController();
+		AFM_SpectatorComponent spectator = AFM_SpectatorComponent.Cast(pc.FindComponent(AFM_SpectatorComponent));
+		if (!spectator)
+			return m_CameraPrefab;
+		
+		bool isLimited = manager.IsLimited() && !spectator.IsSpectatorActive();
+	
+		m_vInitCameraTransform[3] = spectator.GetCameraOriginPos();
+				
+		if (isLimited)
+			return m_LimitedCameraPrefab;
+		else
+			return m_CameraPrefab;
+	}
+	
+	override protected void CreateCamera()
```

Comment:
Have you tried [setting ](https://enfusionengine.com/api/redirect?to=enfusion://ScriptEditor/scripts/Game/Editor/Components/Editor/SCR_CameraEditorComponent.c;83)the init camera transform instead of rewriting this whole logic?

Reply:
Yep, that was my first approach but somewhere in this [section](https://enfusionengine.com/api/redirect?to=enfusion://ScriptEditor/scripts/Game/Editor/Components/Editor/SCR_CameraEditorComponent.c;193) the set position got lost and I have no clue on why.